doc/modules/partial_dependence.rst

```diff
@@ -0,0 +1,124 @@
+    >>> from sklearn.ensemble import GradientBoostingClassifier
+    >>> from sklearn.inspect import plot_partial_dependence
+
+    >>> X, y = make_hastie_10_2(random_state=0)
+    >>> clf = GradientBoostingClassifier(n_estimators=100, learning_rate=1.0,
+    ...     max_depth=1, random_state=0).fit(X, y)
+    >>> features = [0, 1, (0, 1)]
+    >>> fig, axs = plot_partial_dependence(clf, X, features) #doctest: +SKIP
+
+For multi-class classification, you need to set the class label for which
+the PDPs should be created via the ``target`` argument::
+
+    >>> from sklearn.datasets import load_iris
+    >>> iris = load_iris()
+    >>> mc_clf = GradientBoostingClassifier(n_estimators=10,
+    ...     max_depth=1).fit(iris.data, iris.target)
+    >>> features = [3, 2, (3, 2)]
+    >>> fig, axs = plot_partial_dependence(mc_clf, X, features, target=0) #doctest: +SKIP
+
+The same parameter ``target`` is used to specify the target in multi-output
+regression settings.
+
+If you need the raw values of the partial dependence function rather than
+the plots, you can use the :func:`partial_dependence` function::
+
+    >>> from sklearn.inspect import partial_dependence
+
+    >>> pdp, axes = partial_dependence(clf, [0], X=X)
+    >>> pdp  # doctest: +ELLIPSIS
+    array([[ 2.466...,  2.466..., ...
+    >>> axes  # doctest: +ELLIPSIS
+    [array([-1.624..., -1.592..., ...
+
+The values at which the partial dependence should be evaluated are directly
+generated from ``X``. For 2-way partial dependence, a 2D-grid of values is
+generated. The ``values`` field returned by :func:`partial_dependence` gives
+the actual values used in the grid for each target feature. They also
+correspond to the axis of the plots.
+
+For each value of the 'target' features in the ``grid`` the partial
+dependence function needs to marginalize the predictions of the estimator
+over all possible values of the 'complement' features. With the ``'brute'``
+method, this is done by replacing every target feature value of `X` by those
+in the grid, and computing the average prediction.
+
+In decision trees this can be evaluated efficiently without reference to the
+training data (``'recursion'`` method). For each grid point a weighted tree
+traversal is performed: if a split node involves a 'target' feature, the
+corresponding left or right branch is followed, otherwise both branches are
+followed, each branch is weighted by the fraction of training samples that
+entered that branch. Finally, the partial dependence is given by a weighted
+average of all visited leaves. Note that with the ``'recursion'`` method,
+``X`` is only used to generate the grid, not to compute the averaged
+predictions. The averaged predictions will always be computed on the data with
+which the trees were trained.
+
+.. rubric:: Footnotes
+
+.. [1] For classification, the target response is the probability of a class.
+   In particular for binary classification, this is the probability of the
```

Comment:
This footnote is not true for the fast method, right? Maybe clarify here?

Reply:
So you're fine with the fast method and the brute method not plotting the same thing?

Reply:
yes.